x-pack/plugins/observability_solution/dataset_quality/public/components/common/insufficient_privileges.tsx

```diff
@@ -54,7 +54,10 @@ export const PrivilegesWarningIconWrapper = ({
 }) => {
   const [isPopoverOpen, togglePopover] = useToggle(false);
 
-  const handleButtonClick = useCallback(() => togglePopover(true), [togglePopover]);
+  const handleButtonClick = useCallback(
+    () => togglePopover(!isPopoverOpen),
+    [isPopoverOpen, togglePopover]
+  );
```

Comment:
> [!TIP]
> This is equivalent to directly toggling the value, and since `togglePopover` is already a memorized handler, you can just rename it.
```ts
 const handleButtonClick = togglePopover
```